eclipse-theia/theia
# [git] update to latest dugite-extra, dugite-no-gpl

#### What it does

This PR updates `@theia/git` to consume the latest `dugite-extra`, which in turn uses the latest, recently uplifted `dugite-no-gpl`. The later is a fork of upstream `dugite` _(minus GPL-licensed git binaries added at install time)_, that had not been updated in nearly 3 years. 

Update: investigating a `@theia/git` unit test failure (see below), I noticed we needed to update `@theia/git` -> `git-model` -> `GitError enum` to match dugite-no-gpl, that had evolved since the last uplift. See commit message for more details.

#### How to test

Make sure the unit tests pass. Some manual testing of theia git extension would probably be a good idea to make sure nothing was broken.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
